output-comparator/src/test/java/org/mobilitydata/gtfsvalidator/outputcomparator/cli/MainTest.java

```diff
@@ -72,27 +82,59 @@ public void noNewNoticeType_generatesEmptyReport() throws IOException {
 
     writeFile(
         latestNoticeContainer.exportJson(latestNoticeContainer.getValidationNotices()),
-        "src/test/resources/reports/no-new-notices/source-id-1",
-        "latest.json");
+        NO_NEW_NOTICE_FOLDER_NAME,
+        "source-id-1",
+        LATEST_JSON);
     writeFile(
         referenceNoticeContainer.exportJson(referenceNoticeContainer.getValidationNotices()),
-        "src/test/resources/reports/no-new-notices/source-id-1",
-        "reference.json");
+        NO_NEW_NOTICE_FOLDER_NAME,
+        "source-id-1",
+        REFERENCE_JSON);
+
+    JsonObject sourceUrlJsonObject = new JsonObject();
+    sourceUrlJsonObject.addProperty("source-id-1", "url1");
+    sourceUrlJsonObject.addProperty("source-id-2", "url2");
+    sourceUrlJsonObject.addProperty("source-id-3", "url3");
+    writeFile(sourceUrlJsonObject, SOURCE_INFO_FOLDER_NAME, "all", GTFS_LATEST_VERSIONS_JSON);
 
     String[] argv = {
-      "--report_directory", "src/test/resources/reports/no-new-notices",
-      "--new_error_threshold", "1",
-      "--reference_report_name", "reference.json",
-      "--latest_report_name", "latest.json",
-      "--percent_invalid_datasets_threshold", "1",
-      "--output_base", "src/test/resources/output/no-new-notices/acceptance-test-report",
-      "--source_urls", "src/test/resources/gtfs_latest_versions.json"
+      "--percent_corrupted_sources",
+      "5",
+      "--report_directory",
+      tmpDir.getRoot().toPath().resolve(NO_NEW_NOTICE_FOLDER_NAME).toString(),
+      "--new_error_threshold",
+      "1",
+      "--reference_report_name",
+      REFERENCE_JSON,
+      "--latest_report_name",
+      LATEST_JSON,
+      "--percent_invalid_datasets_threshold",
+      "1",
+      "--output_base",
+      tmpDir
```

Comment:
You probably want to use a common method for all these resolves, something like

```
private Path resolve(String... filenames) {
  Path result = tmpDir.getRoot().toPath();
  for (String filename :  filenames) {
    result = result.resolve(filename);
  }
  return result;
}
```

Then it can be used like 

```
"--outpub_base", resolve(NO_NEW_NOTICE_FOLDER_NAME, ACCEPTANCE_TEST_REPORT_FOLDER_NAME),
```

This way we can be sure that we're resolving paths in the same manner.

Also, `writeFile` method should probably take only data and path where to write it, to avoid unnecessary logic duplication (it will still create basedir if required)

Reply:
Good catch! Will be modified in the next commit. Thanks.

> Also, writeFile method should probably take only data and path where to write it, to avoid unnecessary logic duplication (it will still create basedir if required)

👍🏾